File: LeetCode-Problems/Easy/Unique-Morse-Code-Words.py
# Python
class Solution(object):
    def uniqueMorseRepresentations(self, words):
        """
        :type words: List[str]
        :rtype: int
        """
        morse = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        transforms = list()
        differents = list()
        count = 0
        for x in range(0, len(words)):
            convert = ""
            for y in range(0, len(words[x])):
                ascii = ord(words[x][y:y+1])
                ascii = ascii - 97
                convert += morse[ascii]
            transforms.append(convert)
        for x in range(0, len(transforms)):
            if (transforms[x] not in differents):
                count = count + 1
                differents.append(transforms[x])
            else:
                count = count
        return count

File: LeetCode-Problems/Easy/Unique-Morse-Code-Words.java
//Java
import java.util.ArrayList;
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        String[] morse = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        String[] transforms = new String[words.length];
        ArrayList<String> differents = new ArrayList<String>();
        int count = 0;
        for (int i = 0; i < words.length; i++){
            String conversion = "";
            for (int j = 0; j < words[i].length(); j++){
                int asciiLetter = (int) words[i].charAt(j);
                asciiLetter = asciiLetter - 97;
                conversion += morse[asciiLetter];
            }
            transforms[i] = conversion;
        }
        
        for (int i = 0; i < transforms.length; i++){
            if(differents.contains(transforms[i])){
                count = count;
            }
            else{
                differents.add(transforms[i]);
                count++;
            }
        }
        return count;
        
    }
}
